### ISSUE TITLE
v12 - Edge marker/style can be missing in some configuration

### ISSUE THREAD

**Issue by Nicolas-PredictaLab** [xyflow/xyflow#3826]
### Describe the Bug

Link to Discord initial message (video attached) : https://discord.com/channels/771389069270712320/1194252622606106684/1194252622606106684

> I've a strange behaviour after migrating to v12.
You will see in the video attached, if I don't have an already connected edge, adding one with setEdges (ou addEdges too) doesn't add what's seems to be needed to display the arrow marker.

Codesandbox : https://codesandbox.io/p/sandbox/reactflow-dark-mode-forked-2srt8r?file=%2Fsrc%2FFlow.tsx

> I've set some defaultEdgeOptions, marker doesn't seem to be interpretated.
But if you define a fixed option in the A-C edge, like this :
```
  {
    id: "A-C",
    source: "A",
    target: "C",
    markerEnd: {
      type: MarkerType.ArrowClosed,
      color: "red",
    },
  },
  ```
  > You'll see that it make the arrow appear on the first edge! So there is something around this.

### Your Example Website or App

https://codesandbox.io/p/sandbox/reactflow-dark-mode-forked-2srt8r?file=%2Fsrc%2FFlow.tsx

### Steps to Reproduce the Bug or Issue

1. Juste add some nodes and edges and define defaultEdgeOptions

### Expected behavior

I was expecting to work like in V11

### Platform

- OS: macOS
- Browser: Chrome
- Version: "@xyflow/react": "12.0.0-next.4"

**Comment by moklick** [comment]
Thanks! Will be fixed with the next release.

**Comment by moklick** [comment]
fixed in 12.0.0-next.8
